### PR TITLE
state of game

### PR DESCRIPTION
**Pull request by dominik1102**
@sebastian-mikucki , @dominik1102 , @tbradlo , @rmycek please review

### CODE REVIEW

File: src/main/java/com/guysfromusa/carsgame/game_state/GameStateTracker.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import com.guysfromusa.carsgame.game_state.dtos.GameState;
+import com.guysfromusa.carsgame.game_state.dtos.Movement;
+import lombok.NoArgsConstructor;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@NoArgsConstructor
+public class GameStateTracker {
```
Comment (sebastian-mikucki) [inline src/main/java/com/guysfromusa/carsgame/game_state/GameStateTracker.java]: make it component

File: src/main/java/com/guysfromusa/carsgame/game_state/dtos/GameState.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.guysfromusa.carsgame.game_state.dtos;
+
+import lombok.NoArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@NoArgsConstructor
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/game_state/dtos/GameState.java]: this is by default if no other constructor is present

File: src/main/java/com/guysfromusa/carsgame/game_state/GameStateTracker.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@NoArgsConstructor
+@Component
+@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
+public class GameStateTracker {
+    private Map<String, GameState> GAME_STATE = new HashMap<>();
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/game_state/GameStateTracker.java]: concurrent hash map as many threads may access this collection

File: src/main/java/com/guysfromusa/carsgame/game_state/GameStateTracker.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@NoArgsConstructor
+@Component
+@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/game_state/GameStateTracker.java]: singleton is default so pls remove @Scope

File: src/main/java/com/guysfromusa/carsgame/game_state/GameStateTracker.java (hunk 3 above)
Comment (tbradlo) [reply src/main/java/com/guysfromusa/carsgame/game_state/GameStateTracker.java]: it is not a constant, maybe stateByGameId(Map), or gameStateMap etc.

**Pull request merged**